### PR TITLE
Process initial validator set and appHash

### PR DESCRIPTION
**Pull request by adrianbrink**

### CODE REVIEW

**Comment by melekes** [comment]
```
consensus/replay.go:271:25: res.Validators undefined (type *"github.com/tendermint/tendermint/vendor/github.com/tendermint/abci/types".ResponseInitChain has no field or method Validators)
consensus/replay.go:272:13: res.AppHash undefined (type *"github.com/tendermint/tendermint/vendor/github.com/tendermint/abci/types".ResponseInitChain has no field or method AppHash)
```

you can point dep to a specific revision of `abci` in `Gopkg.toml`

**Comment by melekes** [comment]
Can be closed, I think, since original issue was closed.

**Comment by ebuchman** [comment]
We actually still need this so that the app can set the initial validator set

File: consensus/replay.go [hunk 1]
```diff
@@ -264,10 +264,12 @@ func (h *Handshaker) ReplayBlocks(state sm.State, appHash []byte, appBlockHeight
 			Validators:    validators,
 			AppStateBytes: h.appState,
 		}
-		_, err := proxyApp.Consensus().InitChainSync(req)
+		res, err := proxyApp.Consensus().InitChainSync(req)
 		if err != nil {
 			return nil, err
 		}
+		state.Validators = res.Validators
```
Comment (ebuchman) [inline consensus/replay.go]: state is not a pointer so non-mutable

File: consensus/replay.go [hunk 2]
```diff
@@ -264,10 +264,12 @@ func (h *Handshaker) ReplayBlocks(state sm.State, appHash []byte, appBlockHeight
...
+		return res.AppHash, nil
```
Comment (ebuchman) [inline consensus/replay.go]: we still want to run the replay even if we're sending InitChain - the blockchain might have 1000 blocks in it even though the app wants to start from 0 ...

**Comment by ebuchman** [comment]
Thanks - I'll fix this in a separate PR

**Comment by ebuchman** [comment]
See https://github.com/tendermint/tendermint/pull/1701

**Pull request closed without merging**